### PR TITLE
Terror Spider Update, Brown Terrors

### PR DESCRIPTION
**Pull request by Kyep**
New spider type: brown terror spider. This spider has the ability to break open vents, even while inside the vent, but is slow AND weak AND expensive (Queens can breed FOUR normal spiders for the same price). Their purpose is to help the spiders breach into new areas. In the away mission, they also: (A) help prevent Spider accumulation in the Queen nest area making the mission too difficult, (B) stop people treating sealed sections of the mission as permanently safe, (C) allow us to consider making TS their own game mode in future by preventing spiders from being totally shut out by all vents being welded. The brown spider's sprite is new, IK3I made it.

terror_brown
brown_terror_direction_sprites

All TS now see health meters. This is needed for several reasons: (A) Spiders can only wrap corpses, and they need to do so to boost their regeneration/etc, so they need to be able to easily tell if a human is alive or dead. (B) Spider infections drastically change how spiders should treat humanoids, so its important for them to be able to tell if a human is infected with spider eggs or not. (C) Spiders are essentially hunters of humanoids, so it makes sense for them to try to identify and prey on low-health humanoids, rather than attacking blindly. 

Queens also get a new, improved sprite:

new_queen

Coming soon to a dark corner near you:
queen_ambush

Away Mission Changes:
- White spider (which you must kill to unlock bridge, and thus, unlock bosses) is no longer pre-spawned on the away map. Instead, once you start killing her brood, the AI queen will spawn 2 white spiders which roam the map. You have to hunt at least one of these down and kill it to unlock the bridge - and they can be anywhere. 
- Queen will also maintain 4 purples, and 2 browns, at all times after you enter the mission. Previously, Queen could end up with a TON of purples in her room, which made the boss fight semi-impossible for many players. The browns will try to break into areas which you THOUGHT were sealed off and safe... opening the door to other spiders following them.
- Overall, the above changes mean the away mission will be less predictable, as you will no longer be able to rely on the white spider always being in engineering, nor will you be able to rely on your previously-cleared areas staying clear.
- The first gateway room has also been tweaked. There are now enough guns/flashlights for four people, instead of three. The ID Upgrade Machine has also been moved to the middle of the room, to make it more obvious. This was necessary because many people were ignoring it, and it is key to the mission. It allows you to open the doors with your ID, instead of having to tediously hack every single one. 

new_gateway_room

Other Misc Minor Tweaks & Fixes:
- T3 and higher terror spiders no longer drop medals on death. Nobody cared about the medals, and they were kinda gimmicky, so I've removed them.
- TS spiderlings now have a consistent name/description. It is no longer possible to tell TS spiderlings apart from normal spiderlings by their name alone. No more spiderling meta.
- Prying open a welded vent or scrubber now plays an appropriate sound, which people nearby can hear (and be panicked by).
- Using the ventsmash ability while you aren't within range of any welded vent now produces a message to this effect, rather than a pointless progress bar which does nothing.
- Breaking open a welded vent or scrubber now also moves you on top of that welded vent/scrubber. This ensures that a single brown spider cannot break every vent on the station without leaving the safety of the vent network. It also prevents an issue where breaking a vent while you were inside the vent wouldn't update the vent's icon.
- AI-controlled black spiders will no longer jump a huge distance away from you after they bite you. Instead, they will jump a shorter distance, and there will be a clear 'jumps away from' message to indicate what they are doing.
- During the terror spider random event, the initial spiderlings which start the infestation will grow up faster. Additionally, if the 'green' infestation type is randomly chosen (20% chance of this), then the terrors can start with up to 5 greens, rather than up to the previous max of 4. The first change was made to make it less likely that a high-level spider gets killed by an unlucky spawn as a spiderling. The second change was made due to feedback that green type infestations currently feel underwhelming.
- Gray spiders' special ability has been changed. They no longer silence humanoid mobs they attack. Now, they do bonus damage to webbed victims instead. This better achieves the original design idea for the gray spider, that they should be deadly in ambushes but weak outside that.
- Purple spiders' special ability has been changed. They no longer have a RNG-based stun attack. Instead, they have the ability to spin 'thick webs' which block vision and have 2x the health of regular webs. They also move slightly faster.
- The length of the air raid siren which is used to announce TS / level 3 biohazard has been reduced by half.

🆑 Kyep
tweak: Terror Spiders now have a new type in their roster (brown) and see the health status of humanoids (since some of their abilities depend on that status). The Terror Spider away mission has also been adjusted to smooth out areas of no challenge (safe spaces) as well as areas of nigh-impossible challenge (dozens of spiders stacking up together in the final room).
/🆑

### CODE REVIEW

**Comment by KingPhilipIII** [comment]
Time to nest in the Captain's bathroom.

**Comment by Ralta** [comment]
This is a GREAT idea. Terror Spiders are terrifying, but they do end up getting _very_ restricted in where they can travel. Having a specific spider that can facilitate breaking into other areas is awesome - also having a dedicated game mode would be awesome too :D

**Comment by Citinited** [comment]
The few times I've played as terror spiders have always been hampered by welded vents that were welded due to an earlier antagonist, so I'm all for this change.

**Comment by KingPhilipIII** [comment]
Another thing being only Reds, Purples and the Queen(Who shouldn't be out and about) can force doors open meaning other ones were forced to follow them around or risk being trapped.

Not including the Prince and the Mother, who are their own batch of horrifying and rarely appear.

**Comment by KasparoVy** [comment]
Lmao, nice name. Not quite what I'd call a brown terror but I suppose it'll do.

**Comment by KasparoVy** [comment]
Brown terrors happen after eating the spiciest food you can stuff in your face.
These are more like 'disconnect in ten seconds or less', but that's endemic to terror spiders in general.

I dig the mechanics, though.

**Comment by Necaladun** [comment]
lgtm, also love the sprites

**Comment by KingPhilipIII** [comment]
While we’re on the topic of monster spiders, am I the only one that thinks Princes have the least matching sprite of them stylistically? I understand it needs to be obvious that you’re dealing with the single most dangerous individual you can(Stat wise, not even accounting for player skill) but there’s a disconcerting feeling every time I look at it.

**Comment by Kyep** [comment]
The Prince sprite looks different from all the others. It was made by a different artist, with a different design idea. It was actually originally made for the black spider, and looked like this:
black_old
Suffice it to say, I removed the red hourglass, and used it for the prince instead, because its just that scary:
black_new
I think it works well, A black spider with glowing red eyes that looks like its coming at you out of the darkness is terrifying. Exactly what the prince (who is essentially a one-man army) should be.
I've been trying to have spider sprites be based on their physical features/abilities, but there's a real shortage of spriters. IK3L has been great, but prior to this PR, I was actually operating with placeholder sprites for some types (still, to this day) and

**Comment by KingPhilipIII** [comment]
I'm not saying the sprite doesn't work, or that it's not fitting. Definitely inspires some terror to see it charging at me. But it does not fit in with the others is ultimately my point.

**Comment by Kyep** [comment]
That's intentional. Its actually really hard for all the sprites to 'fit in' together given that they've been made by a series of different artists each with their own completely different design ideas.

So, there will always be some sprites that look quite different to others. Personally, I don't feel this is such a bad thing. It allows the sprites to have a more unique feel. It also allows me to use the most different sprites for bosses/minibosses.... which underlines their danger nicely. You know a creature is dangerous when its sprite is very difference (and more menacing) than others of its kind.

**Comment by KingPhilipIII** [comment]
I guess I can see your point, it still strikes me as off however, and I understand making them fit in together isn't an option but a lack of solutions does not change a perceived problem.

**Comment by Kyep** [comment]
I can always ask IK3I to resprite the prince... but IMHO it would be a real shame to stop using that prince sprite, its just so.... SCARY. Yeah it looks different, but its just so good.

**Comment by KingPhilipIII** [comment]
They were the product of your mind, so it's ultimately your choice, again I'm not saying the sprite doesn't work I was just getting more details on why it was different so that was why I brought it up.

**Comment by KingPhilipIII** [comment]
Also semi-related but out of the scope of this PR, have you considered getting audio bits made for terror spooders? Some good spooky stuff to go with some of their more distinctive stuff (Like when the Queen matures it says she shrieks) or maybe some chittering noises.

**Comment by Kyep** [comment]
Audio bits would be nice.
I don't know anyone with the necessary skills, and who I could persuade to make them, though.
I could always go through some free libraries of re-usable sounds, and look for sounds from there.

The main issue with, for example, giving the Queen's screech a distinctive sound is that she does it at the start of her nesting process... and this being a distinctive sound would instantly give away her position before the nest is even established. This would lead to a quick death for the nest as soon as people learned what that sound meant.

So, if I was going to start adding sounds to TS, I would start by adding them for things like spider bite attacks. Things that happen, would be heard by lots of people, and be scary, but which don't give crew a massive advantage if they hear them.

**Comment by KingPhilipIII** [comment]
Yea I get it I just used that as an example. Alternatively just make itso the screech is only audible in that room or adjacent ones to minimize risk of 'GUYS I JUST HEARD SOME BAD SHIT'

**Comment by Kyep** [comment]
Sounds are distance-based, I think, not room-based. Also queen is never going to nest in a room with other people in it, so room-based sounds would only be heard by her.

**Comment by KingPhilipIII** [comment]
+1 on the new Queen sprite. It’s darker pallete makes it blend better too.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm [hunk 1]
```diff
@@ -140,81 +146,60 @@
...
-							DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/green, 2, 1)
+						if(ai_nest_is_full())
+							neststep = 4
 						else
-							DoLayTerrorEggs(/mob/living/simple_animal/hostile/poison/terror_spider/purple, 2, 0)
-						neststep = 6
+							var/obj/structure/spider/eggcluster/terror_eggcluster/N = locate() in get_turf(src)
+							if(!N)
+								spider_lastspawn = world.time
+								var num_purple = CountSpidersType(/mob/living/simple_animal/hostile/poison/terror_spider/purple)
```
Comment (tigercat2000) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm]: You forgot the slashes.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 /obj/structure/spider/spiderling/terror_spiderling
 	name = "spiderling"
-	desc = "A fast-moving tiny spider, prone to making aggressive hissing sounds. Hope it doesn't grow up."
+	desc = "It moves very quickly, hissing loudly for its size... and it has disproportionately large fangs. Hopefully it does not grow up..."
```
Comment (tigercat2000) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm]: The old description was better.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm (hunk 1 above)
Comment (Kyep) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm]: Fixed

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm (hunk 2 above)
Comment (Kyep) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm]: Reverted

**Comment by Kyep** [comment]
@tigercat2000 Changes done

**Pull request merged**